packages/web/src/shared/components/TypeaheadInput.tsx

```diff
@@ -0,0 +1,149 @@
+import { ComponentProps, useEffect, useState } from 'react';
+import { useCombobox } from 'downshift';
+import TextInput from './TextInput';
+import { Icon } from './Icon';
+
+export interface TypeaheadInputItem {
+  label: string;
+  value: string;
+}
+
+export interface TypeaheadInputProps
+  extends Omit<ComponentProps<'input'>, 'value' | 'onChange'> {
+  value?: string;
+  items: TypeaheadInputItem[];
+  onChange(value?: string): void;
+  onCreate?(text?: string): void;
+}
+
+/**
+ * Text input with autocomplete suggestions.
+ *
+ * Suggestions have the shape of `{ label: string; value: string; }`.
+ * When an item is selected, its value will be passed to the `onChange` event.
+ *
+ * To support the creation of new items, attach the `onCreate` event.
+ * The text value will be passed to the event.
+ */
+export default function TypeaheadInput({
```

Comment:
![image](https://github.com/arrocke/gloss-translation/assets/17209828/6cbe983e-4e38-468c-9d94-c737cc6091a9)

Something isn't quite right with z-index it seems.